crates/settings/src/order.rs

```diff
@@ -70,31 +72,35 @@ impl OrderString {
             })
             .transpose()?;
 
-        let inputs = self
-            .inputs
-            .into_iter()
-            .map(|input| {
-                tokens
-                    .get(&input)
-                    .ok_or(ParseOrderStringError::TokenParseError(
-                        ParseTokenStringError::NetworkNotFoundError(input.clone()),
-                    ))
-                    .map(Arc::clone)
-            })
-            .collect::<Result<Vec<_>, _>>()?;
-
-        let outputs = self
-            .outputs
-            .into_iter()
-            .map(|output| {
-                tokens
-                    .get(&output)
-                    .ok_or(ParseOrderStringError::TokenParseError(
-                        ParseTokenStringError::NetworkNotFoundError(output.clone()),
-                    ))
-                    .map(Arc::clone)
-            })
-            .collect::<Result<Vec<_>, _>>()?;
+        let mut inputs = vec![];
```

Comment:
let's keep the implementation using iter map rather than for loops -- much clearer and easier to read.

Reply:
cant really do this on an item, because need to check the network match on each input